File: app/src/main/java/android/rr/apksapp/SplashActivity.java
package android.rr.apksapp;

import android.content.Intent;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class SplashActivity extends AppCompatActivity {
    Handler handler;
    Runnable runnable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        handler = new Handler();

        runnable = new Runnable() {
            @Override
            public void run() {
                startActivity(new Intent(SplashActivity.this, Dashboard.class));
                SplashActivity.this.finish();
            }
        };
    }

    @Override
    protected void onResume() {
        super.onResume();

        handler.postDelayed(runnable, 1500);
    }

    @Override
    protected void onStop() {
        super.onStop();

        handler.removeCallbacks(runnable);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        if (null != handler)
            handler = null;

        if (null != runnable)
            runnable = null;
    }

}

File: app/src/main/java/android/rr/apksapp/SplashActivity.kt
package android.rr.apksapp

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.support.v7.app.AppCompatActivity

class SplashActivity : AppCompatActivity() {
    var handler: Handler? = null
    var runnable: Runnable? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        handler = Handler()
        runnable = Runnable {
            startActivity(Intent(this@SplashActivity, Dashboard::class.java))
            finish()
        }
    }

    override fun onResume() {
        super.onResume()
        handler?.postDelayed(runnable, 1500)
    }

    override fun onStop() {
        super.onStop()
        handler?.removeCallbacks(runnable)
    }

    override fun onDestroy() {
        super.onDestroy()
        if (null != handler) handler = null
        if (null != runnable) runnable = null
    }
}